### PR TITLE
[Civl] Anonymous actions in refinement specification of yield procedures

### PR DESCRIPTION
**Pull request by shazqadeer**
It is occasionally tedious to invent names of atomic actions whose only purpose is to serve as refinement specification of a yield procedure. This PR enables such specifications to be specified via anonymous actions. The new test file added in this PR provides a good illustration.

### CODE REVIEW

**Comment by menesro** [comment]
Yes, I like it. It is also easier to read the file because the atomic specification is close to the yield procedure.

File: Test/civl/large-samples/shared-vector.bpl [hunk 1]
```diff
@@ -305,32 +305,26 @@ type Tid;
 type Mutex = Option Tid;
 var {:layer 0, 9} MutexPool: Map Loc Mutex;
 
-right action {:layer 1, 9} Atomic_Mutex_Alloc() returns ({:linear} one_loc_l: One Loc)
-modifies MutexPool;
+yield procedure {:layer 0} Mutex_Alloc() returns ({:linear} one_loc_l: One Loc);
+refines right action {:layer 1, 9} _
```
[the remark this answers is not in the document]
Comment (shazqadeer) [reply Test/civl/large-samples/shared-vector.bpl:309]: It is so hard that I don't even know how to do it right now :-). The problem is that I need to allow attributes as well as the action body both of which begin with ```{```.  Take a look at BoogiePL.atg when you find some time.  In any case, the convention of using ```_``` for the name of the anonymous action feels nice to me.

**Comment by shazqadeer** [comment]
The current behavior is that the name of the anonymous action could be an arbitrary string (including ```_```) but it is thrown away and an internally generated name is used. There is one other option: use the supplied name if it is different from ```_``` but generate an internal name if it is ```_```.

Any preference?

**Comment by bkragl** [comment]
> There is one other option: use the supplied name if it is different from `_` but generate an internal name if it is `_`.

This sounds like a good solution to me that will avoids potential confusion. `_` is commonly used for anonymous things in programming languages, and being allowed to refer to the action as if it were declared separately seems reasonable.

**Pull request merged**